test/test_service.rb

```diff
@@ -273,4 +273,63 @@ def test_patch_service
       data['metadata']['annotations']['key'] == 'value'
     end
   end
+
+  def test_json_patch_service
+    service = Kubeclient::Resource.new
+    name = 'my_service'
+
+    service.metadata = {}
+    service.metadata.name      = name
+    service.metadata.namespace = 'development'
+
+    stub_core_api_list
+    expected_url = "http://localhost:8080/api/v1/namespaces/development/services/#{name}"
+    stub_request(:patch, expected_url)
+      .to_return(body: open_test_file('service_patch.json'), status: 200)
+
+    patch = [
+      { 'op' => 'replace', 'path' => '/metadata/annotations/key1', 'value' => 'value' },
+      { 'op' => 'add', 'path' => '/metadata/annotations/key2', 'value' => 'value' }
```

Comment:
in principle, once you use functionally different patch here, ideally you'd mock a different return result than `service_patch.json`.  If you don't want that, nor want to patch only `/metadata/annotations/key`, please add a comment on the `to_return` saying it's inaccurate result for this patch.